### PR TITLE
feat(dato-tooltip): add the ability to reinitiate tooltip on text change

### PR DESCRIPTION
**Pull request by orgilor**

### CODE REVIEW

File: lib/src/directives/tooltip.directive.ts [hunk 1]
```diff
@@ -191,4 +166,53 @@ export class DatoTooltipDirective implements OnDestroy, AfterViewInit {
...
+      this.triggerOff = fromEvent(this.host.nativeElement, off)
+        .pipe(untilDestroyed(this))
+        .subscribe(() => {
+          this.hide();
+        });
+    }
+  }
+
+  private unsubscribe() {
+    if (this.triggerOn) {
```
Comment (bh86) [inline lib/src/directives/tooltip.directive.ts]: this.triggerOn && this.triggerOn.unsubscribe();

File: playground/src/app/components-gallery/previews/tooltip-preview/tooltip-preview.component.html [hunk 2]
```diff
@@ -122,16 +122,26 @@
...
   <ul style="max-width: 100px;">
-    <li class="ellipsis" datoTooltip="Long Long All Text" datoTooltipPosition="right" datoTooltipOnOverflow="true" style="display: inline-block">Long Long All Text</li>
-    <li class="ellipsis" datoTooltip="Not visible" datoTooltipPosition="right" datoTooltipOnOverflow="true" style="display: inline-block">Short</li>
+    <li class="ellipsis" [datoTooltip]="longText" datoTooltipPosition="right" [datoTooltipOnTextOverflow]="true"
+        style="display: inline-block">
+      {{longText}}
+    </li>
+    <li class="ellipsis" [datoTooltip]="editableTextControl.value" datoTooltipPosition="right"
+        [datoTooltipOnTextOverflow]="true" style="display: inline-block">
+      <span-expanded [formControl]="editableTextControl"></span-expanded>
```
Comment (bh86) [inline playground/src/app/components-gallery/previews/tooltip-preview/tooltip-preview.component.html]: what's wrong with an input/textarea? :)

Comment (orgilor) [reply playground/src/app/components-gallery/previews/tooltip-preview/tooltip-preview.component.html]: nothing, its just doesnt cover my usecase as good

File: lib/src/directives/tooltip.directive.ts [hunk 3]
```diff
@@ -85,46 +93,13 @@ export class DatoTooltipDirective implements OnDestroy, AfterViewInit {
   constructor(private host: ElementRef, private iconRegistry: IconRegistry) {}
 
   ngAfterViewInit() {
+    this.viewInit = true;
+
     if (this.datoTooltipTrigger === 'manual') {
```
Comment (shaharkazaz) [inline lib/src/directives/tooltip.directive.ts]: maybe write instead:
`if (this.datoTooltipTrigger !== 'manual') { this.initTooltip() }`

**Pull request merged**